erdpy_network/transaction_completion_strategy.py

```diff
@@ -16,6 +27,36 @@ def is_completed(self, transaction: ITransactionOnNetwork) -> bool:
         return not transaction.get_status().is_pending()
 
 
+# this class is similar to the one in erdjs-network-providers
 class TransactionCompletionStrategyOnProxy:
-    def is_completed(self, transaction: ITransactionOnNetwork):
-        raise NotImplementedError('Method is not implemented yet')
+    def is_completed(self, transaction: ITransactionOnNetwork) -> bool:
+        if transaction.get_status().is_pending():
+            return False
+
+        for event in KNOWN_COMPLETION_EVENTS:
+            if transaction.logs.find_first_or_none_event(event):
+                return True
+
+        if self.__is_certainly_move_balance(transaction.data):
+            return transaction.status.is_executed()
+
+        #  Imprecise condition, uncertain completion (usually sufficient, though).
+        #  This is WRONG when (at least): timeOf(block with execution at destination is notarized) < timeOf(the "completedTxEvent" occurs).
+        if transaction.hyperblock_nonce > 0:
+            return True
+
+        return False
+
+    def __is_certainly_move_balance(self, transaction_data: str) -> bool:
```

Comment:
Perhaps add a comment with a link towards the erdjs implementation, as well?

https://github.com/ElrondNetwork/elrond-sdk-erdjs-network-providers/blob/main/src/transactionCompletionStrategy.ts#L50

Reply:
Added!